### PR TITLE
update multistream and multiplex

### PR DESCRIPTION
**Pull request by whyrusleeping**
Updates go-stream-muxer to use the latest interface changes from go-multiplex, and also embeds the lazy connection changes from go-multistream

### CODE REVIEW

File: Godeps/_workspace/src/github.com/whyrusleeping/go-multiplex/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+*.swp
```
Comment (jbenet) [inline Godeps/_workspace/src/github.com/whyrusleeping/go-multiplex/.gitignore]: garbage

File: Godeps/_workspace/src/github.com/whyrusleeping/go-multiplex/.gxlastpubver [hunk 2]
```diff
@@ -0,0 +1 @@
+QmYmW76Y7NxvWwW3GPUmpbHd4mm1iW14f5UimeJ9UoofEs
```
Comment (jbenet) [inline Godeps/_workspace/src/github.com/whyrusleeping/go-multiplex/.gxlastpubver:1]: nice!

File: Godeps/_workspace/src/github.com/whyrusleeping/go-multiplex/package.json [hunk 3]
```diff
@@ -0,0 +1,5 @@
+{
+  "name": "go-multiplex",
+  "version": "1.0.0",
```
Comment (jbenet) [inline Godeps/_workspace/src/github.com/whyrusleeping/go-multiplex/package.json:3]: boom!

File: multiplex/multiplex.go [hunk 4]
```diff
@@ -29,15 +29,19 @@ func (c *conn) OpenStream() (smux.Stream, error) {
...
 func (c *conn) Serve(handler smux.StreamHandler) {
-	c.Multiplex.Serve(func(s *mp.Stream) {
-		handler(s)
-	})
+	for {
+		s, err := c.AcceptStream()
+		if err != nil {
+			return
+		}
+		go handler(s)
```
Comment (jbenet) [inline multiplex/multiplex.go:43]: wheeeee!

**Comment by jbenet** [comment]
LGTM

File: Godeps/_workspace/src/github.com/whyrusleeping/go-multiplex/.gxlastpubver (hunk 2 above)
Comment (whyrusleeping) [inline Godeps/_workspace/src/github.com/whyrusleeping/go-multiplex/.gxlastpubver:1]: hrm, that might only be resolveable while my computer is online... but its a start

Comment (jbenet) [inline Godeps/_workspace/src/github.com/whyrusleeping/go-multiplex/.gxlastpubver:1]: @ipfsbot pin QmYmW76Y7NxvWwW3GPUmpbHd4mm1iW14f5UimeJ9UoofEs

o/ that should work

**Pull request merged**